backend/services/deviceauth/api/http/routing.go

```diff
@@ -38,27 +77,94 @@ func supportsMethod(method string, methods []string) bool {
 
 // Automatically add OPTIONS method support for each defined route,
 // only if there's no OPTIONS handler for that route yet
-func AutogenOptionsRoutes(routes []*rest.Route, gen HttpOptionsGenerator) []*rest.Route {
+func AutogenOptionsRoutes(router *gin.Engine, gen HttpOptionsGenerator) {
 
+	routes := router.Routes()
 	methodGroups := make(map[string][]string, len(routes))
 
 	for _, route := range routes {
-		methods, ok := methodGroups[route.PathExp]
+		if strings.HasPrefix(route.Path, "/api/internal") {
+			continue
+		}
+		methods, ok := methodGroups[route.Path]
 		if !ok {
 			methods = make([]string, 0)
 		}
 
-		methodGroups[route.PathExp] = append(methods, route.HttpMethod)
+		methodGroups[route.Path] = append(methods, route.Method)
 	}
 
-	options := make([]*rest.Route, 0, len(methodGroups))
 	for route, methods := range methodGroups {
 		// skip if there's a handler for OPTIONS already
 		if !supportsMethod(http.MethodOptions, methods) {
-			options = append(options,
-				rest.Options(route, gen(methods)))
+			router.OPTIONS(route, gen(methods))
 		}
 	}
 
-	return append(routes, options...)
+}
+
+func init() {
+	if mode := os.Getenv(gin.EnvGinMode); mode != "" {
+		gin.SetMode(mode)
+	} else {
+		gin.SetMode(gin.ReleaseMode)
+	}
+	gin.DisableConsoleColor()
+}
+
+func NewRouter(app devauth.App, db store.DataStore) http.Handler {
+	router := gin.New()
+	router.Use(accesslog.Middleware())
+	router.Use(requestid.Middleware())
+
+	d := NewDevAuthApiHandlers(app, db)
+
+	publicAPIs := router.Group(".")
+	publicAPIs.Use(identity.Middleware())
+
+	mgmtAPIV2 := publicAPIs.Group(apiUrlManagementV2)
+	devicesAPIs := publicAPIs.Group(apiUrlDevicesV1)
```

Comment:
The auth_requests API does not use the identity middleware as there's no bearer token in the request.
```suggestion
	devicesAPIs := router.Group(apiUrlDevicesV1)
```